tests/transform_dialect/matmul_fill_spec_pack_peel.mlir

```diff
@@ -39,153 +38,196 @@ module attributes { transform.with_named_sequence } {
+    // Peel the for loop
+    %for_ops = transform.structured.match ops{["scf.for"]} in %variant_op : (!transform.any_op) -> !transform.op<"scf.for">
+    %inner_for, %outer_for = transform.split_handle %for_ops : (!transform.op<"scf.for">) -> (!transform.op<"scf.for">, !transform.op<"scf.for">)
+    %peeled_outer_loop, %remainder = transform.loop.peel %outer_for {peel_front = true}
       : (!transform.op<"scf.for">) -> (!transform.any_op, !transform.any_op)
+
+    // Clean up.
     transform.include @cleanup failures(propagate) (%variant_op) : (!transform.any_op) -> ()
 
-    // Find the fill and the second forall operations.
-    %fused_fill_1 = transform.structured.match ops{["linalg.fill"]} in %variant_op : (!transform.any_op) -> !transform.any_op
-    %fill_consumer = transform.get_consumers_of_result %fused_fill_1[0] : (!transform.any_op) -> (!transform.any_op)
+    // Find the fill operation to fuse
+    %fill_op = transform.structured.match ops{["linalg.fill"]} in %variant_op : (!transform.any_op) -> !transform.any_op
+    // Get the consumers of the fill. It has to be a scf.forall op.
+    %peeled_loop = transform.get_consumers_of_result %fill_op[0] : (!transform.any_op) -> (!transform.op<"scf.forall">)
 
-    // Fuse fill operation into the forall loop
-    %fused_fill_2, %__ = transform.structured.fuse_into_containing_op %fused_fill_1 into %fill_consumer
-      : (!transform.any_op, !transform.any_op) -> (!transform.any_op, !transform.any_op)
+    // Fuse the fill within the loop.
+    %peel_fused, %13 = transform.structured.fuse_into_containing_op %fill_op into %peeled_loop : (!transform.any_op, !transform.op<"scf.forall">) -> (!transform.any_op, !transform.any_op)
 
     // Clean up.
     transform.include @cleanup failures(propagate) (%variant_op) : (!transform.any_op) -> ()
-    transform.iree.eliminate_empty_tensors %variant_op : (!transform.any_op) -> ()
 
     // Bufferize and drop HAL decriptor from memref ops.
-    %variant_op_3 = transform.iree.bufferize %variant_op : (!transform.any_op) -> !transform.any_op
+    transform.iree.eliminate_empty_tensors %variant_op : (!transform.any_op) -> ()
+    %14 = transform.iree.bufferize %variant_op : (!transform.any_op) -> !transform.any_op
+
     transform.yield
   }
 }
 
-// CHECK-LABEL: @matmul_example
-//       CHECK: memref.alloc() : memref<1x1x8x4x4x8xi32, 2>
-//       CHECK: memref.alloc() : memref<1x1x8x8x8x8xi8, 2>
-//       CHECK: memref.alloc() : memref<1x1x8x4x4x8xi8, 2>
-//       CHECK: memref.alloc() : memref<1x4x16x64xi32, 1>
-//       CHECK: memref.alloc() : memref<1x4x64x64xi8, 1>
-//       CHECK: memref.alloc() : memref<1x1x16x64xi8, 1>
+// CHECK-LABEL: @matmul_i32
+//       CHECK: memref.alloc() : memref<1x1x4x4x8x8xi32, 2>
+//       CHECK: memref.alloc() : memref<1x1x4x8x4x8xi32, 2>
+//       CHECK: memref.alloc() : memref<1x1x256x64xi32, 1>
+//       CHECK: memref.alloc() : memref<1x1x64x256xi32, 1>
+//       CHECK: memref.alloc() : memref<1x1x8x16x4x8xi32, 2>
```

Comment:
It allocates `memref<1x1x8x16x4x8xi32, 2>` instead of `memref<1x1x4x8x4x8xi32, 2>` for outputs in L1. Then it generates and fills a memref.subview for the required output.